src/mapping/query/function/split.rs

```diff
@@ -0,0 +1,247 @@
+use super::prelude::*;
+use bytes::Bytes;
+use lazy_static::lazy_static;
+use std::collections::BTreeMap;
+use std::ops::Deref;
+
+#[derive(Debug)]
+pub(in crate::mapping) struct SplitFn {
+    path: Box<dyn Function>,
+    pattern: Box<dyn Function>,
+    limit: Option<Box<dyn Function>>,
+}
+
+impl SplitFn {
+    #[cfg(test)]
+    pub(in crate::mapping) fn new(
+        path: Box<dyn Function>,
+        pattern: Box<dyn Function>,
+        limit: Option<Box<dyn Function>>,
+    ) -> Self {
+        Self {
+            path,
+            pattern,
+            limit,
+        }
+    }
+}
+
+lazy_static! {
+    static ref I_FLAG: Value = Value::from(Bytes::from("i"));
+    static ref G_FLAG: Value = Value::from(Bytes::from("g"));
+    static ref M_FLAG: Value = Value::from(Bytes::from("m"));
+}
```

Comment:
Can't we use `char`/`u8` here?

E.g., use the `Bytes::from_static` const method, and potentially not use `lazy_static` but call `Value::from` at the relevant location (the overhead would be non-existent).